### PR TITLE
Some improvement to the advanced digital signature module

### PR DESCRIPTION
**Pull request by drskalman**
This will improve the advanced digital signature modules with few changes:
- Making the categories of multi sigs more logical and assign more accurate names.
- Correct few facts about BLS signature.
- Add images for DKG and Ring signatures.
-

### CODE REVIEW

File: syllabus/1-Cryptography/6-Advanced_Signatures-slides.md [hunk 1]
```diff
@@ -36,8 +36,10 @@ Certificate transparency: [explanation](https://certificate.transparency.dev/how
 
 ### Certificates in Web3
 
-We are building systems that don't have a "Certificate Authority".<br/>
-But we can still use certificates in some niche instances.
+We are building systems that do not have a centralized "Certificate Authority".<br/>
+But we can still use certificates:
+ - Certified by a distributed authority.
```
Comment (naterarmstrong) [inline syllabus/1-Cryptography/6-Advanced_Signatures-slides.md:41]: I think it is very important to get into the nuance on this topic if you mention it at this point. The main reason is that with a certificate authority, any and all certificates can be checked (largely) offline.[^1] However, in a distributed authority, there _can be no single key_. 

For instance, If you want to trust something signed by the validators of a blockchain in a trustless way, you need to sync the whole chain up to the current state in some way, and then make sure that you get signatures from a secure majority of the current validators. Or make sure that the multisignature address (as that's much more likely than individual sigs) has been confirmed on the chain in some way, etc.

 It's much more complex to use a certificate from a distributed authority than a CA.

I do like the other changes though, I agree niche was a bit excessive. 

[^1]: Not technically entirely true, due to things like checking databases for revocation certificates, checking the certificate transparency databases, etc. But fundamentally there is an offline chain of trust from the CA's hardcoded keys to the key providing them to you.

File: syllabus/1-Cryptography/6-Advanced_Signatures-slides.md [hunk 2]
```diff
@@ -66,13 +71,13 @@ We often want signatures that must be signed<br/>by multiple parties to become v
 
 <pba-flex center>
 
-- Verifier enforced
-- Cryptographic threshold
-- Cryptographic non-threshold<br/>(a.k.a. signature aggregation)
+- Trivial: union of individual signatures.
```
Comment (naterarmstrong) [inline syllabus/1-Cryptography/6-Advanced_Signatures-slides.md:74]: Union is often a confusing word for those not mathematically inclined. Can just call it 'sending multiple signatures'.

File: syllabus/1-Cryptography/6-Advanced_Signatures-slides.md [hunk 3]
```diff
@@ -95,7 +100,7 @@ Even in a web3 system, the verifier can be _distinct_ from the blockchain. 5 peo
 
 ### Cryptographic Multi-Sigs
 
-We want a succinct way to demonstrate that everyone from some set of parties have signed a message. This is achieved purely on the signer side (without support from the verifier).
+We want a succinct way to demonstrate that everyone from some set of parties have signed a message. This could be achieved purely on the signer side (without support from the verifier).
```
Comment (naterarmstrong) [inline syllabus/1-Cryptography/6-Advanced_Signatures-slides.md:103]: Good catch

File: syllabus/1-Cryptography/6-Advanced_Signatures-slides.md [hunk 4]
```diff
@@ -190,7 +201,7 @@ However...
 
 - DKGs remain tricky (for threshold).
 - Soft key derivations are typically insecure for BLS.
-- Verifiers are hundreds of times slower than Schnorr, due to using pairings, for a single signature.
+- Verifiers are significantly slower than Schnorr, due to using pairings, for a single signature.
```
Comment (naterarmstrong) [inline syllabus/1-Cryptography/6-Advanced_Signatures-slides.md:204]: 👍

File: syllabus/1-Cryptography/6-Advanced_Signatures-slides.md [hunk 5]
```diff
@@ -202,6 +213,7 @@ Schnorr & BLS multi-signatures avoid complicating verifier logic,<br/>but introd
 - DKG protocols
 - Reduced key derivation ability
 - Verification speed
+- Proof of possession verification.
```
Comment (naterarmstrong) [inline syllabus/1-Cryptography/6-Advanced_Signatures-slides.md:216]: I'm assuming this is referring to the part of the musig protocol that avoids chosen-public key attacks? If so, I would just cut this as it is already covered in the communication-heavy DKG protocol.

Comment (drskalman) [reply syllabus/1-Cryptography/6-Advanced_Signatures-slides.md:216]: Even for simple aggregation (when you don't need to run DKG) with BLS signatures you still can forge other participants signature if you don't show PoP.

Comment (naterarmstrong) [reply syllabus/1-Cryptography/6-Advanced_Signatures-slides.md:214]: Ah ok, good to leave it in then.

**Review by Asamartino — APPROVED** [review]

**Pull request merged**